File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        hashmap={}

        for ind,num in enumerate(nums):
            diff=target-num

            if diff in hashmap:
                return [hashmap[diff],ind]
            
            hashmap[num]=ind
        return []

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.rs
impl Solution {
    pub fn product_except_self(nums: Vec<i32>) -> Vec<i32> {

        let mut result = vec![1; nums.len()] ;

        for (ind ,num ) in nums.iter().enumerate(){
            for (ind2 ,num2 ) in nums.iter().enumerate(){
                if ind2!= ind {
                    result[ind] *= num2;

                }
            }
        }
        result



    }
}
